test/units/playbook/test_collectionsearch.py

```diff
@@ -0,0 +1,38 @@
+# (c) 2020 Ansible Project
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.playbook.collectionsearch import CollectionSearch
+
+import pytest
+
+
+def test_collection_static_warning(capsys):
+    """Test that collection name is not templated.
+
+    Also, make sure that users see the warning message for the referenced name.
+    """
+
+    collection_name = 'foo.{{bar}}'
+    cs = CollectionSearch()
+    assert collection_name in cs._load_collections(None, [collection_name])
```

Comment:
What does that method return? Can we have an equality check here?
Also, is there a public method that could be called instead of the private one?

Reply:
It returns a list of collection names and is adding 'ansible.legacy' to that list. I'm not aware of a public method to call instead.

Reply:
If the order is deterministic, you could have a more precise check here.

Reply:
Well, in this test case, it should always be the first element based on the current code. But in this case, we don't actually care about positioning, just that it appears anywhere in the list, independent of which other collections are inserted. Do you feel always checking against the first element is better? That could open it up to test failures if we later decide to alter the ordering for some reason that I cannot yet foresee.

Reply:
I think either way is fine. But in general, I prefer tests to be as precise as possible to prevent some buggy behavior from squeezing in just because it has similar properties.